adafruit_displayio_layout/widgets/dial.py

```diff
@@ -0,0 +1,1096 @@
+
+    :param int needle_width: requested pixel width of the triangular needle
+    :param int needle_color: color value for the needle, defaults to red (0xFF0000)
+
+    :param int tick_color: tick line color (24-bit hex value)
+    :param int major_ticks: number of major ticks
+    :param int major_tick_stroke: major tick line stroke width, in pixels
+    :param int major_tick_length: major tick length, in pixels
+    :param str major_tick_labels: array of strings for the major tick labels
+    :param float tick_label_scale: the scaling of the tick labels, default = 1.0
+    :param Font tick_label_font: font to be used for major tick labels
+    :param int tick_label_color: color for the major tick labels
+    :param Boolean angle_tick_labels: set True to rotate the major tick labels to
+     match the tick angle
+
+    :param int minor_ticks: number of minor ticks (per major tick)
+    :param int minor_tick_stroke: minor tick line stroke width, in pixels
+    :param int minor_tick_length: minor tick length, in pixels
+
+    :param int background_color: background color (RGB tuple
+     or 24-bit hex value), set None for transparent
+
+
+    :param (float,float) anchor_point: (X,Y) values from 0.0 to 1.0 to define the dial's
+     anchor point relative to the dial's bounding box
+    :param (int,int) anchored_position: (x,y) pixel value for the location
+     of the `anchor_point`
+
+
+    **Simple example of dial and moving needle**
+
+    See file: ``examples/displayio_layout_dial_simpletest.py``
+
+
+
+    .. figure:: dial.gif
+       :scale: 100 %
+       :figwidth: 50%
+       :align: center
+       :alt: Diagram of the dial widget with needle in motion.
+
+       This is a diagram of a dial widget with the needle moving from its
+       minimum to maximum positions.
+
+    .. figure:: dial_variables_angles.png
+       :scale: 50 %
+       :figwidth: 70%
+       :align: center
+       :alt: Diagram showing the definition of ``start_angle`` and ``sweep_angle``,
+        both are in units of degrees.
+
+       Diagram showing the definition of ``start_angle`` and ``sweep_angle``,
+       both are in units of degrees.
+
+    .. figure:: dial_variables_min_max_values.png
+       :scale: 50 %
+       :figwidth: 70%
+       :align: center
+       :alt: Diagram showing the defintion of ``min_value`` and ``max_value``.
+
+       Diagram showing the defintion of ``min_value`` and ``max_value``.
+
+    .. figure:: dial_variables_ticks.png
+       :scale: 50 %
+       :figwidth: 70%
+       :align: center
+       :alt: Diagram showing the various parameters for setting the dial labels
+        and major and minor tick marks.
+
+       Diagram showing the various parameters for setting the dial labels
+       and major and minor tick marks.
+    """
+
+    # The dial is a subclass of Group->Widget.
+
+    def __init__(
+        self,
+        width=100,
+        height=100,
+        padding=0,  # keepout amount around border, in pixels
```

Comment:
I think this should default to something like `12` or so. At `0` it seems to cut off parts of the labels when a dial is created with minimal arguments passed. At `12` the minimal argument dial is looking nice.

Reply:
Updated per your suggestion.